File: frontend/next/src/dashboard/api/personality.ts
import { api } from "./client";

export interface AIPersonalityResponse {
  id: string;
  business_id: string;
  host_name: string | null;
  brand_voice: string | null;
  focus_areas: string | null;
  avoid_topics: string | null;
}

export interface AIPersonalityUpdate {
  host_name?: string | null;
  brand_voice?: string | null;
  focus_areas?: string | null;
  avoid_topics?: string | null;
}

export function fetchPersonality(token: string) {
  return api.get<AIPersonalityResponse | null>("/api/dashboard/personality", token);
}

export function upsertPersonality(token: string, payload: AIPersonalityUpdate) {
  return api.put<AIPersonalityResponse>("/api/dashboard/personality", payload, token);
}

File: backend/app/schemas/personality.py
import uuid

from pydantic import BaseModel


class AIPersonalityUpdate(BaseModel):
    host_name: str | None = None
    brand_voice: str | None = None
    focus_areas: str | None = None
    avoid_topics: str | None = None


class AIPersonalityResponse(BaseModel):
    id: uuid.UUID
    business_id: uuid.UUID
    host_name: str | None = None
    brand_voice: str | None = None
    focus_areas: str | None = None
    avoid_topics: str | None = None

    model_config = {"from_attributes": True}
